domain/domain.go

```diff
@@ -612,14 +613,21 @@ func (do *Domain) autoAnalyzeWorker(lease time.Duration) {
 		log.Warnf("[stats] campaign owner fail:", errors.ErrorStack(err))
 	}
 	statsHandle := do.StatsHandle()
+	analyzeTicker := time.NewTicker(lease)
+	defer analyzeTicker.Stop()
 	for {
-		if statsOwner.IsOwner() {
-			err := statsHandle.HandleAutoAnalyze(do.InfoSchema())
-			if err != nil {
-				log.Error("[stats] auto analyze fail:", errors.ErrorStack(err))
+		select {
+		case <-analyzeTicker.C:
+			if statsOwner.IsOwner() {
+				err := statsHandle.HandleAutoAnalyze(do.InfoSchema())
```

Comment:
`HandleAutoAnalyze` takes very long time.
We should be about to quit in the middle of `HandleAutoAnalyze` if `domain` exit.

Reply:
The `HandleAutoAnalyze` will analyze at most one table/index in one time, but currently, we cannot exit in the middle of an `analyze`.

Reply:
You can pass domain.exit channel, it would be closed when domain is closed.